src/main/java/agh/ics/oop/RectangularMap.java

```diff
@@ -0,0 +1,85 @@
+package agh.ics.oop;
+
+import java.util.ArrayList;
+import java.util.List;
+public class RectangularMap implements IWorldMap {
+    private final int width;
+    private final int height;
+
+    private List<Animal> animals = new ArrayList<>();
+    public RectangularMap(int width,int height) {
+        this.width = width;
+        this.height = height;
+    }
+    public List<Animal> getAnimals()
+    {
+        return animals;
+    }
+    @Override
+    public boolean canMoveTo(Vector2d position) {
+        if (!isOccupied(position)){
+            if (position.x <= width && position.y <= height && position.x >= 0 && position.y >=0)
+            {
+                return true;
+            }
+
+        }
+        return false;
+    }
+
+    @Override
+    public boolean place(Animal animal) {
+        if (animals.size() >= width*height)
+        {
+            return false;
+        }
+        animals.add(animal);
+        return true;
+
+    }
+
+    @Override
+    public boolean isOccupied(Vector2d position) {
+        for(Animal x: animals )
+        {
+            if (x.isAt(position)){
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public Object objectAt(Vector2d position) {
+        for(Animal x: animals)
+        {
+            if (x.isAt(position)){
+                return x;
+            }
+        }
+        return null;
+    }
+    public void moveAnimals(MoveDirection[] moves)
```

Comment:
fajny pomysł żeby przesuwać zwierzęta nie tworząc dodatkowej ich listy, ale trochę dodajemy tej klasie odpowiedzialność silnika do poruszania - to on steruje i mówi, że np. przechodzimy po wszystkich zwierzątkach modulo. Także raczej umieściłbym to w silniku. Można ewentualnie dodać do mapy metodę `getAnimals()`, co rozwiąże problem dostępu do zwierząt.